scripts/language_model/model/qa.py

```diff
@@ -0,0 +1,351 @@
+        p_mask : NDArray or None, shape(batch_size, seq_length)
+
+        Returns
+        -------
+        x : NDarray, shape(batch_size, seq_length)
+            Masked end logits.
+        """
+        F = mx.ndarray
+        if not self._eval:
+            start_states = F.gather_nd(
+                hidden_states,
+                F.concat(
+                    F.contrib.arange_like(hidden_states,
+                                          axis=0).expand_dims(1),
+                    start_positions.expand_dims(
+                        1)).transpose())  # shape(bsz, hsz)
+            start_states = start_states.expand_dims(1)
+            start_states = F.broadcast_like(
+                start_states, hidden_states)  # shape (bsz, slen, hsz)
+        x = self.dense_0(F.concat(hidden_states, start_states, dim=-1))
+        x = self.layernorm(x)
+        x = self.dense_1(x).squeeze(-1)
+        if p_mask is not None and self._eval:
+            p_mask = p_mask.expand_dims(-1)
+            p_mask = F.broadcast_like(p_mask, x)
+        if p_mask is not None:
+            x = x * (1 - p_mask) - 1e30 * p_mask
+        return x
+
+
+class XLNetPoolerAnswerClass(Block):
+    """ Compute SQuAD 2.0 answer class from classification and start tokens hidden states. """
+    def __init__(self, units=768, dropout=0.1, prefix=None, params=None):
+        super(XLNetPoolerAnswerClass, self).__init__(prefix=prefix,
+                                                     params=params)
+        with self.name_scope():
+            self._units = units
+            self.dense_0 = nn.Dense(units,
+                                    in_units=2 * units,
+                                    activation='tanh',
+                                    use_bias=True,
+                                    flatten=False)
+            self.dense_1 = nn.Dense(1,
+                                    in_units=units,
+                                    use_bias=False,
+                                    flatten=False)
+            self._dropout = nn.Dropout(dropout)
+
+    def __call__(self, hidden_states, start_states=None, cls_index=None):
+        # pylint: disable=arguments-differ
+        return super(XLNetPoolerAnswerClass,
+                     self).__call__(hidden_states, start_states, cls_index)
+
+    def forward(self, hidden_states, start_states, cls_index):
+        # pylint: disable=arguments-differ
+        """Get answerability logits from the model output and start states.
+
+        Parameters
+        ----------
+        hidden_states : NDArray, shape (batch_size, seq_length, hidden_size)
+        start_states : NDArray, shape (batch_size, hidden_size)
+            Typically weighted average hidden_states along second dimension.
+        cls_index : NDArray, shape (batch_size)
+            Index of [CLS] token in sequence.
+
+        Returns
+        -------
+        x : NDarray, shape(batch_size,)
+            CLS logits.
+        """
+        F = mx.ndarray
```

Comment:
It would be good to add a comment regarding what is blocking hybridization of this Block, or to hybridize it